test/core/test_k8s.py

```diff
@@ -188,14 +188,14 @@ def test_create_deployment(self, create_p):
 
         create_p.assert_called_once()
 
-        create_p.side_effect = MagicMock(side_effect=ApiException(500, 'Test'))
+        create_p.side_effect = MagicMock(side_effect=ApiException(422, 'Test'))
         with self.assertRaises(api.ApiError) as e:
             k8s.create_deployment(pvc, 'test')
 
-        expected = ("Error when creating the deployment deployment: (500)\n"
+        expected = ("Error when creating the deployment deployment: (422)\n"
```

Comment:
"deployment deployment" sounds like a typo (I know it is not). Maybe:

```suggestion
        expected = ("Error when creating the deployment object: (422)\n"
```